### ISSUE TITLE
Enhancement: [prefer-destructuring] add option to check enums

### ISSUE THREAD

**Issue by pillowfication** [typescript-eslint/typescript-eslint#10294]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-destructuring/

### Description

Currently the `@typescript-eslint/prefer-destructuring` rule has options to turn on/off checking when destructuring against arrays and objects. Is it possible to add an option to check destructuring against enums? Although enums are objects, destructuring them is often awkward and unwanted.

### Fail

```TypeScript
enum Direction { Up, Right, Down, Left }

// { "object": true, "enum": true }
// ❌ Use object destructuring.
const dir = Direction.Up
```

### Pass

```TypeScript
enum Direction { Up, Right, Down, Left }

// { "object": true, "enum": true }
const { Up: dir } = Direction

// { "object": true, "enum": false }
const dir = Direction.Up
```

**Comment by bradzacher** [comment]
To do this we would require type information to detect that thee thing is an enum.
Adding type info to an existing rule greatly restricts the usage of the rule because many codebases cannot/do not use type info.

As this usecase is very, very, very niche -- it is not worth gating the rule behind type information to catch it. Thus we shall reject this proposal.
